From 626dded557de1e7b90cb847df9e900d40be5af1a Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@tuxdriver.com>
Date: Wed, 14 Dec 2011 07:09:07 -0500
Subject: [PATCH 1/8] Add sample irqbalance environment file

It was pointed out that the example systemd unit file pointed to a corresponding
environment file that had no sample.  Fix that up, and modify the unit file to
pass available option via environment variables rather than command line options
since that looks a little cleaner.

Signed-off-by: Neil Horman <nhorman@tuxdriver.com>

add irqbalance args variable to env file

Allow users to pass general arguments to irqbalance through systemd

Signed-off-by: Neil Horman <nhorman@tuxdriver.com>
---
 misc/irqbalance.env     | 26 ++++++++++++++++++++++++++
 misc/irqbalance.service |  5 ++---
 2 files changed, 28 insertions(+), 3 deletions(-)
 create mode 100644 misc/irqbalance.env

diff --git a/misc/irqbalance.env b/misc/irqbalance.env
new file mode 100644
index 0000000..bd87e3d
--- /dev/null
+++ b/misc/irqbalance.env
@@ -0,0 +1,26 @@
+# irqbalance is a daemon process that distributes interrupts across
+# CPUS on SMP systems.  The default is to rebalance once every 10
+# seconds.  This is the environment file that is specified to systemd via the 
+# EnvironmentFile key in the service unit file (or via whatever method the init
+# system you're using has. 
+#
+# ONESHOT=yes
+#    after starting, wait for a minute, then look at the interrupt
+#    load and balance it once; after balancing exit and do not change
+#    it again.
+#IRQBALANCE_ONESHOT=
+
+#
+# IRQBALANCE_BANNED_CPUS
+#    64 bit bitmask which allows you to indicate which cpu's should
+#    be skipped when reblancing irqs.  Cpu numbers which have their 
+#    corresponding bits set to one in this mask will not have any
+#    irq's assigned to them on rebalance
+#
+#IRQBALANCE_BANNED_CPUS=
+
+#
+# IRQBALANCE_ARGS
+#    append any args here to the irqbalance daemon as documented in the man page
+#
+#IRQBALANCE_ARGS=
diff --git a/misc/irqbalance.service b/misc/irqbalance.service
index f349616..aae2b03 100644
--- a/misc/irqbalance.service
+++ b/misc/irqbalance.service
@@ -3,9 +3,8 @@ Description=irqbalance daemon
 After=syslog.target
 
 [Service]
-EnvironmentFile=/etc/sysconfig/irqbalance
-Type=forking
-ExecStart=/usr/sbin/irqbalance $ONESHOT
+EnvironmentFile=/path/to/irqbalance.env
+ExecStart=/usr/sbin/irqbalance $IRQBALANCE_ARGS
 
 [Install]
 WantedBy=multi-user.target
-- 
1.7.11.4

